src/runtime/audience-action-local-flow.ts

```diff
@@ -302,13 +302,14 @@ export class AudienceActionLocalFlow implements AudienceActionFlow {
     // Hide prompt before closing the prompt.
     setImportantStyles(this.wrapper_, {'opacity': '0'});
     // Wait for form submit request to send before closing the prompt.
-    await this.delay_(1000);
     this.eventManager_.logSwgEvent(
       AnalyticsEvent.ACTION_BYOP_NEWSLETTER_OPT_IN_SUBMIT
     );
+    await this.complete_();
+
+    await this.delay_(1000);
```

Comment:
is this delay still needed?

Reply:
It is possible. However, I don't think we can intercept a third party post call & it's response, especially we are expanding to multiple thrid parties, I don't think there is an unanimous signal from the different ESP tools.

We could potentially wait for a longer period of time if we think 1 second if not enough?